presto-main/src/main/java/com/facebook/presto/server/WebUiResource.java

```diff
@@ -46,22 +48,33 @@
 
 import static com.google.common.io.Resources.getResource;
 import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.Response.Status.MOVED_PERMANENTLY;
+import static javax.ws.rs.core.UriBuilder.fromPath;
 
 @Path("/")
-public class QueryExecutionResource
+public class WebUiResource
 {
     // synthetic task id used by the output buffer of the top task
     private static final TaskId OUTPUT_TASK_ID = new TaskId("output_buffer", 0, 0);
 
     private final QueryManager manager;
 
     @Inject
-    public QueryExecutionResource(QueryManager manager)
+    public WebUiResource(QueryManager manager)
     {
         requireNonNull(manager, "manager is null");
         this.manager = manager;
     }
 
+    @GET
+    @Path("/{path: .*}")
```

Comment:
This will be called for the endpoints that are not matched by other resources (/v1/info for example won't go through this). All static resources should go through this resource and they will be directed to /ui/{path}. 
The full file name with the extension will be used by the `ClassPathResourceHandler` eventually to load the resource.